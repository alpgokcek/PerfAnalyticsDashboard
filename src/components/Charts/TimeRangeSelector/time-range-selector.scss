@import "../../../styles/placeholders.scss";

.time-range-selector {
  display: grid;
  grid-template-columns: 1fr 1fr;
  column-gap: 40px;
  max-width: 1024px;
  margin: 0 auto;
  @include phone {
    grid-template-columns: 1fr;
    row-gap: 40px;
  }
  &__container {
    font-weight: 700;
    div {
      margin-bottom: 5px;
      text-align: center;
    }
  }
}
