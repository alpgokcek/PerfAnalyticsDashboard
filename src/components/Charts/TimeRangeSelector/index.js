import React from "react";
import PropTypes from "prop-types";
import { DateTimePickerComponent } from "@syncfusion/ej2-react-calendars";

import "./time-range-selector.scss";

const TimeRangeSelector = (props) => {
  const { onChange, filters } = props;

  const handleDateChange = (e, isStartDate = false) => {
    if (isStartDate) onChange({ ...filters, start: e.target.value });
    else onChange({ ...filters, end: e.target.value });
  };

  return (
    <div className="time-range-selector">
      <div className="time-range-selector__container">
        <div>Start Date</div>
        <DateTimePickerComponent
          data-test="datetime-start"
          placeholder="Choose a start date and time"
          value={filters.start}
          onChange={(e) => handleDateChange(e, true)}
          format="dd-MMM-yy HH:mm"
          step={30}
        ></DateTimePickerComponent>
      </div>

      <div className="time-range-selector__container">
        <div>End Date</div>
        <DateTimePickerComponent
          data-test="datetime-end"
          placeholder="Choose an end date and time"
          value={filters.end}
          onChange={(e) => handleDateChange(e)}
          format="dd-MMM-yy HH:mm"
          step={30}
        ></DateTimePickerComponent>
      </div>
    </div>
  );
};

TimeRangeSelector.propTypes = {
  onChange: PropTypes.func,
  filters: PropTypes.object,
};

export default TimeRangeSelector;
