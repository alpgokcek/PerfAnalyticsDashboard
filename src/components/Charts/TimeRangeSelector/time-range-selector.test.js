import React from "react";
import TimeRangeSelector from "./index";

import Enzyme, { mount } from "enzyme";
import toJson from "enzyme-to-json";
import EnzymeAdapter from "enzyme-adapter-react-16";
import { findByTestAttribute } from "../../../test";

Enzyme.configure({ adapter: new EnzymeAdapter() });

const defaultProps = {
  onChange: jest.fn,
  filters: { start: new Date("2021-05-23"), end: new Date("2021-05-23") },
};

const setup = (props) => {
  return mount(<TimeRangeSelector {...props} />);
};

describe("snapshot and unit tests for TimeRangeSelector", () => {
  let wrapper;
  beforeAll(() => {
    const mGetRandomValues = jest.fn().mockReturnValueOnce(new Uint32Array(10));
    Object.defineProperty(window, "crypto", {
      value: { getRandomValues: mGetRandomValues },
    });
    jest
      .useFakeTimers("modern")
      .setSystemTime(new Date("2021-05-23").getTime());
  });

  beforeEach(() => {
    wrapper = setup(defaultProps);
  });

  it("should render correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should show the current date/time", () => {
    const startDate = findByTestAttribute(wrapper, "datetime-start");
    expect(startDate.find("input").instance().value).toMatch("23-May-21 03:00");
    const endDate = findByTestAttribute(wrapper, "datetime-end");
    expect(endDate.find("input").instance().value).toMatch("23-May-21 03:00");
  });

  it("should update the date/time after filter change", () => {
    wrapper.setProps({
      ...defaultProps,
      filters: {
        start: new Date("2021-05-21"),
        end: new Date("2021-05-22"),
      },
    });
    const startDate = findByTestAttribute(wrapper, "datetime-start");
    expect(startDate.find("input").instance().value).toMatch("21-May-21 03:00");
    const endDate = findByTestAttribute(wrapper, "datetime-end");
    expect(endDate.find("input").instance().value).toMatch("22-May-21 03:00");
  });
});
