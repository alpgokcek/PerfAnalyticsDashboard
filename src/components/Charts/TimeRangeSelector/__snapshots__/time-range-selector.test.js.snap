// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot and unit tests for TimeRangeSelector should render correctly 1`] = `
<TimeRangeSelector
  filters={
    Object {
      "end": 2021-05-23T00:00:00.000Z,
      "start": 2021-05-23T00:00:00.000Z,
    }
  }
  onChange={[Function]}
>
  <div
    className="time-range-selector"
  >
    <div
      className="time-range-selector__container"
    >
      <div>
        Start Date
      </div>
      <n
        data-test="datetime-start"
        format="dd-MMM-yy HH:mm"
        onChange={[Function]}
        placeholder="Choose a start date and time"
        step={30}
        value={2021-05-23T00:00:00.000Z}
      >
        <input
          data-test="datetime-start"
          key="1"
        />
      </n>
    </div>
    <div
      className="time-range-selector__container"
    >
      <div>
        End Date
      </div>
      <n
        data-test="datetime-end"
        format="dd-MMM-yy HH:mm"
        onChange={[Function]}
        placeholder="Choose an end date and time"
        step={30}
        value={2021-05-23T00:00:00.000Z}
      >
        <input
          data-test="datetime-end"
          key="2"
        />
      </n>
    </div>
  </div>
</TimeRangeSelector>
`;
